stl/inc/ranges

```diff
@@ -3831,13 +3843,24 @@ namespace ranges {
 
             _NODISCARD constexpr decltype(auto) operator[](const difference_type _Idx) const
                 noexcept(noexcept(_STD get<_Index>(*(_Current + _Idx)))) /* strengthened */
-                requires random_access_range<_Base> {
+                requires random_access_range<_Base>&& is_reference_v<range_reference_t<_Base>> {
 #if _ITERATOR_DEBUG_LEVEL != 0
                 _Verify_offset(_Idx);
 #endif // _ITERATOR_DEBUG_LEVEL != 0
                 return _STD get<_Index>(*(_Current + _Idx));
             }
 
+            _NODISCARD constexpr decltype(auto) operator[](const difference_type _Idx) const
+                noexcept(is_nothrow_move_constructible_v<tuple_element_t<_Index, range_reference_t<_Base>>> //
+                        && noexcept(_STD get<_Index>(*(_Current + _Idx)))) /* strengthened */
+                requires random_access_range<_Base> {

```

Comment:
Minor throughput question. Given that I did not implement the `get_element` helper we will have both versions of `begin` and `operator[]` synthesized.

Should we add a `&& (!is_reference_v<range_reference_t<_Base>>)`  and above?

Reply:
I'm not concerned about it, but if I was I would suggest doing something CPO-ish instead: I'm opposed to complicating our requires-clauses as a mechanism to avoid the cost of subsumption checking.

If you _do_ want to do something, please do so in a followup PR since this one is already being ported.